pkg/kademlia/dialer_test.go

```diff
@@ -203,15 +203,15 @@ func TestSlowDialerHasTimeout(t *testing.T) {
 			peer := peer
 			group.Go(func() error {
 				_, err := dialer.PingNode(ctx, peer.Local().Node)
-				require.Error(t, err, context.DeadlineExceeded)
-				require.True(t, transport.Error.Has(err))
-
+				if !transport.Error.Has(err) || errs.Unwrap(err) != context.DeadlineExceeded {
+					return errs.New("invalid error: %v", err)
+				}
```

Comment:
i'm still not totall clear why this wouldn't work: 
```
require.Equal(t, errs.Unwrap(err), context.DeadlineExceeded)
require.True(t, transport.Error.Has(err))
```

Reply:
Because `group.Go` starts things in a goroutine. And any method call on `*testing.T` results in a data race.

Reply:
Oh, I think what you wrote would work, and I over-matched on a heuristic about causing `FailNow` to happen in non-test goroutines. That's been an endless source of bugs for me, so I try to avoid it.